xbmc/utils/log.h

```diff
@@ -166,24 +166,20 @@ class CLog : public ISettingsHandler, public ISettingCallback
                                                   const wchar_t* format,
                                                   Args&&... args)
   {
-    GetInstance().FormatAndLogInternal(
-        level, StringUtils::Format(L"{0:s}: {1:s}", functionName, format).c_str(),
-        std::forward<Args>(args)...);
+    GetInstance().FormatAndLogInternal(level,
+                                       StringUtils::Format(L"{0:s}: {1:s}", functionName, format),
+                                       std::forward<Args>(args)...);
   }
 
-  template<typename Char, typename... Args>
+  template<typename... Args>
   inline void FormatAndLogInternal(spdlog::level::level_enum level,
-                                   const Char* format,
+                                   std::string format,
                                    Args&&... args)
   {
-    // TODO: for now we manually format the messages to support both python- and printf-style formatting.
-    //       this can be removed once all log messages have been adjusted to python-style formatting
-    auto logString = StringUtils::Format(format, std::forward<Args>(args)...);
-
     // fixup newline alignment, number of spaces should equal prefix length
-    StringUtils::Replace(logString, "\n", "\n                                                   ");
+    StringUtils::Replace(format, "\n", "\n                                                   ");
```

Comment:
I'm wondering if there is a way to get rid of doing this explicitely so that we can simply use `spdlog` without any additional processing.

I see the following options:

1. We provide a newline helper method in `CLog` which as to be explicitely used when using a newline in a log message. That would mean we would have to track down all log messages using a newline.
2. We implement a custom `spdlog` formatter (see https://github.com/gabime/spdlog/wiki/3.-Custom-formatting and https://github.com/gabime/spdlog/blob/v1.x/include/spdlog/formatter.h) which does this for us without having to do it manually for every log message in `CLog`. That would also allow us to use `spdlog` loggers directly and still use newlines.

Reply:
Yea a custom formatter would be good